shm/Map.hpp

```diff
@@ -0,0 +1,44 @@
+#pragma once
+
+#include <random>
+#include <vector>
+#include <algorithm>
+#include <iostream>
+#include "Island.hpp"
+
+class Map
+{
+
+public:
+    Map() : mapXsize(10), mapYsize(11) { randomIslandGenerator(10); }
+    Map(int islandsAmount) { randomIslandGenerator(islandsAmount); }
+    Map(int sizeY, int sizeX, int islandsAmount) : mapXsize(sizeY), mapYsize(sizeX)
+    {
+        randomIslandGenerator(islandsAmount);
+    }
+
+    std::vector<Island> getIslands() { return islands; }
+    Island getIsland(const Coordinates& corr)
+    {
+        return *(std::find_if(std::begin(islands), std::end(islands), [corr](Island& is) {return is.getPosition() == corr; }));
+    }
```

Comment:
move to cpp

Reply:
Additionally I think it will be easier to use this function later if you return pointer to an Island instead of Island, additionally we will have some performance gain, will be no need to copy the whole object, just a pointer.